### PR TITLE
Add donating code guide

### PR DESCRIPTION
**Pull request by chris-crone**
As discussed in the 2020-02-19 community meeting, I've written down a brief guide for how to contribute code to the CNAB project.

Please give feedback as this is just a first pass!

### CODE REVIEW

File: docs/donating-code.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Donating Code to the CNAB Project
+
+The CNAB project has a history of collaboration across companies and countries
+on the specification and its associated code bases. The
+[CNAB GitHub organization](https://github.com/cnabio) is home to the
+specification itself as well as reference implementations and helper libraries.
+Much of the code found there was developed by one company or person and then
+donated to the CNAB project. This document details what kind of code projects
+are is accepted into the CNAB project and what the process for donating it is.
```
Comment (radu-matei) [inline docs/donating-code.md]: "what kind of code projects _are_ accepted"

**Review by radu-matei — APPROVED** [review]
LGTM

**Review by technosophos — APPROVED** [review]
LGTM as-is.

We might have to ask JDF to weigh in (at some point) on whether patents usage must be granted if the license is MIT.

File: docs/donating-code.md (hunk 1 above)
Comment (chris-crone) [reply docs/donating-code.md]: Nice catch, thanks :)

**Pull request merged**